lib/trento/activity_log.ex

```diff
@@ -83,6 +84,87 @@ defmodule Trento.ActivityLog do
     from(l in q, where: l.type not in @user_management_log_types)
   end
 
+  defp maybe_search_by_metadata(query, params) do
+    maybe_metadata_search_string = params[:search]
+
+    is_valid_search_string? =
+      maybe_metadata_search_string != nil &&
+        maybe_metadata_search_string != "" &&
+        String.match?(maybe_metadata_search_string, ~r/^[a-zA-Z0-9-_\*\s]+$/)
+
+    case is_valid_search_string? &&
+           parse_metadata_search_string(maybe_metadata_search_string) do
+      parsed_query when is_binary(parsed_query) ->
+        from q in query,
+          select: %{
+            id: q.id,
+            metadata: q.metadata,
+            m0: fragment("jsonb_path_query(?, ?)", q.metadata, ^parsed_query),
+            type: q.type,
+            actor: q.actor,
+            inserted_at: q.inserted_at,
+            updated_at: q.updated_at
+          }
+
+      _ ->
+        query
+    end
+  end
+
+  defp parse_metadata_search_string(search_string) do
+    "#{get_search_selector(search_string)} #{get_search_predicate(search_string)}"
+  end
+
+  defp get_search_selector(_), do: "$"
+
+  @search_regex "[a-zA-Z0-9_-]"
+  defp get_search_predicate(search_string) do
+    search_words = String.split(search_string, " ")
+    add_surrounding_brackets? = length(search_words) > 1
+
+    joiner =
+      case Enum.any?(search_words, fn word -> word == "AND" or word == "OR" end) do
+        true ->
+          " "
+
+        _ ->
+          " || "
+      end
+
+    joined_string =
+      Enum.map_join(search_words, joiner, &word_handler/1)
+
+    case add_surrounding_brackets? do
+      true ->
+        "? (#{joined_string})"
+
+      _ ->
+        "? #{joined_string}"
+    end
```

Comment:
This will be addressed/made cleaner/more extensible (to the extent possible) in another PR.

Reply:
Addessed in [52e3fd3d](https://github.com/trento-project/web/pull/3059/commits/52e3fd3d390713f41f9fdd59f037c171239b56c1).